server/sonar-server/src/main/java/org/sonar/server/computation/issue/BaseIssuesLoader.java

```diff
@@ -0,0 +1,105 @@
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.computation.issue;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.ibatis.session.ResultContext;
+import org.apache.ibatis.session.ResultHandler;
+import org.sonar.api.rule.RuleKey;
+import org.sonar.api.rule.RuleStatus;
+import org.sonar.api.utils.log.Loggers;
+import org.sonar.core.issue.DefaultIssue;
+import org.sonar.core.issue.db.IssueDto;
+import org.sonar.core.issue.db.IssueMapper;
+import org.sonar.core.persistence.DbSession;
+import org.sonar.core.persistence.MyBatis;
+import org.sonar.server.computation.batch.BatchReportReader;
+import org.sonar.server.computation.component.TreeRootHolder;
+import org.sonar.server.db.DbClient;
+
+import static com.google.common.collect.FluentIterable.from;
+import static org.sonar.core.rule.RuleKeyFunctions.stringToRuleKey;
+
+/**
+ * Loads all the project open issues from database, including manual issues.
+ *
+ */
+public class BaseIssuesLoader {
+
+  private final Set<RuleKey> activeRuleKeys;
+  private final TreeRootHolder treeRootHolder;
+  private final DbClient dbClient;
+  private final RuleRepository ruleRepository;
+
+  public BaseIssuesLoader(BatchReportReader reportReader, TreeRootHolder treeRootHolder,
+    DbClient dbClient, RuleRepository ruleRepository) {
+    this.activeRuleKeys = from(reportReader.readMetadata().getActiveRuleKeyList()).transform(stringToRuleKey()).toSet();
+    this.treeRootHolder = treeRootHolder;
+    this.dbClient = dbClient;
+    this.ruleRepository = ruleRepository;
+  }
+
+  public List<DefaultIssue> loadForComponentUuid(String componentUuid) {
+    DbSession session = dbClient.openSession(false);
+    final List<DefaultIssue> result = new ArrayList<>();
+    try {
+      Map<String, String> params = ImmutableMap.of("componentUuid", componentUuid);
+      session.select(IssueMapper.class.getName() + ".selectNonClosedByComponentUuid", params, new ResultHandler() {
+        @Override
+        public void handleResult(ResultContext resultContext) {
+          DefaultIssue issue = ((IssueDto) resultContext.getResultObject()).toDefaultIssue();
+
+          // TODO this field should be set outside this class
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)